internal/controller/controller.go

```diff
@@ -262,7 +265,12 @@ func aiGatewayRouteToAttachedGatewayIndexFunc(o client.Object) []string {
 		ret = append(ret, fmt.Sprintf("%s.%s", ref.Name, aiGatewayRoute.Namespace))
 	}
 	for _, ref := range aiGatewayRoute.Spec.ParentRefs {
-		ret = append(ret, fmt.Sprintf("%s.%s", ref.Name, aiGatewayRoute.Namespace))
+		// Use the namespace from parentRef if specified, otherwise use the route's namespace.
+		namespace := aiGatewayRoute.Namespace
+		if ref.Namespace != nil && *ref.Namespace != "" {
+			namespace = string(*ref.Namespace)
+		}
+		ret = append(ret, fmt.Sprintf("%s.%s", ref.Name, namespace))
```

Comment:
sorry curious why is this change relevant here? if it's purely the coverage purpose, I would appreciate it if you could do it in another PR... I would prefer clear context in each commit.